app/human-detector-app/auth/KeyCloakAuth.ts

```diff
@@ -0,0 +1,92 @@
+import { makeRedirectUri, useAuthRequest } from 'expo-auth-session';
+import * as AuthSession from 'expo-auth-session';
+import * as CryptoJS from 'crypto-js';
+
+function genCodeVerifier(size: number) {
+  const chars = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890';
+  const charLength = chars.length;
+  let result = '';
+  for (let i = 0; i < size; i++) {
+    result += chars.charAt(Math.floor(Math.random() * charLength));
+  }
+  return result;
+}
+
+/**
+ * KeyCloak instance class to hold a KeyCloak auth instance. This object
+ * will hold values that will be important to keep track of.
+ *
+ * To get the authorization token, we need to make sure that codeVerifier and
+ * redirectUri don't change after being given a value.  These are used for verification of authorization.
+ */
+export default class KeyCloakInstance {
```

Comment:
Maybe out of scope for this PR, but will this class provide a way to refresh our credentials?

Reply:
I wanted to code the flow of the app before I get to that but that's what I'm planning on right now.  I just want to make sure that the stuff around it will make sense for that or maybe I'll have to do something else if that makes sense?

There's some things I have to do with the app after signing in that hasn't been worked on yet since it will involve sign up with user info and stuff like that.